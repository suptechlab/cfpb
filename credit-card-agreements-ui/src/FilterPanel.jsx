import './FilterPanel.less'
import Agreement from './Filters/Agreement'
import { connect } from 'react-redux'
import Issuer from './Filters/Issuer'
import React from 'react'

export class FilterPanel extends React.Component {
  render() {
    const descPublicResponse = "The company's optional public-facing " +
      "response to a consumer's complaint"
    const descConsumerConsent = 'Whether a consumer opted to publish their ' +
      'compaint narrative'
    const descTags = 'Data that supports easier searching and sorting of ' +
      'complaints submitted by or on behalf of older Americans and/or ' +
      'servicemembers'

    return (
      <section className="filter-panel">
        <h3>Filter results by...</h3>
        <Agreement />
        <hr />
        <Issuer />
      </section>
    )
  }
}

const mapStateToProps = state => ( {
  aggs: state.aggs
} )

export default connect( mapStateToProps )( FilterPanel )
