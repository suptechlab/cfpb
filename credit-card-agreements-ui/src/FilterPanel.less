@import (less) "base.less";

.filter-panel {
  padding: @gutter-normal;

  @media @desktop {
    padding: @grid_gutter-width;
  }

  border: 1px solid @gray-40;

  background: @gray-5;

  hr {
    margin-top: @gutter-normal;
    margin-bottom: @gutter-normal;

    color: @gray-40;
  }

  .clear {
    margin-left: @gutter-normal;
  }

  ul {
    padding-left: 0;
  }

  li {
    overflow: hidden;
    padding-box: 5px;
  }

}
