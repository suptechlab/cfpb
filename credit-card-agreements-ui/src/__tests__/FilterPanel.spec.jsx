import React from 'react';
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import FilterPanel from '../FilterPanel';
import renderer from 'react-test-renderer';

function setupSnapshot() {
  const middlewares = [thunk]
  const mockStore = configureMockStore(middlewares)
  const store = mockStore({
    query: {},
    aggs: {}
  })

  return renderer.create(
    <Provider store={store}>
      <FilterPanel />
    </Provider>
  )
}

describe('initial state', () => {
  it('renders without crashing', () => {
    const target = setupSnapshot()
    const tree = target.toJSON()
    expect(tree).toMatchSnapshot()
  });
});

