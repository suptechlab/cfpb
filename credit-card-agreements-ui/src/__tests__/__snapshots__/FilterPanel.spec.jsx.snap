// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial state renders without crashing 1`] = `
<section
  className="filter-panel"
>
  <h3>
    Filter results by...
  </h3>
  <section>
    <div
      className="o-expandable_header o-expandable_target"
    >
      <h4
        className="o-expandable_header-left o-expandable_label"
      >
        Credit card name
      </h4>
    </div>
    
    <section
      className="typeahead "
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="m-btn-inside-input input-contains-label"
      >
        <label
          className="input-contains-label_before input-contains-label_before__search"
        />
        <input
          autoComplete="off"
          className="a-text-input"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Enter card name"
          type="text"
          value=""
        />
      </div>
    </section>
    <ul />
  </section>
  <hr />
  <section>
    <div
      className="o-expandable_header o-expandable_target"
    >
      <h4
        className="o-expandable_header-left o-expandable_label"
      >
        Issuer name
      </h4>
    </div>
    
    <section
      className="typeahead "
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <div
        className="m-btn-inside-input input-contains-label"
      >
        <label
          className="input-contains-label_before input-contains-label_before__search"
        />
        <input
          autoComplete="off"
          className="a-text-input"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Enter card issuer name"
          type="text"
          value=""
        />
      </div>
    </section>
    <ul />
  </section>
</section>
`;
